src/ToricVarieties/AlgebraicCycles/attributes.jl

```diff
@@ -0,0 +1,218 @@
+
+Return the exponents of the algebraic cycle `ac`.
+
+# Examples
+```jldoctest
+julia> dP2 = del_pezzo(2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> d = ToricDivisor(dP2, [1,2,3,4,5])
+A torus-invariant, non-prime divisor on a normal toric variety
+
+julia> ac = AlgebraicCycle(d)
+An algebraic cycle on a normal toric variety given by 6V(x3)+V(e1)+7V(e2)
+
+julia> exponents(ac)
+[0   0   1   0   0]
+[0   0   0   1   0]
+[0   0   0   0   1]
+```
+"""
+exponents(ac::AlgebraicCycle) = matrix(ZZ,[k for k in exponent_vectors(polynomial(ac).f)])
+export exponents
+
+
+@doc Markdown.doc"""
+    polynomial(ac::AlgebraicCycle)
+
+Return the polynomial in the Chow ring of the normal
+toric variety `toric_variety(ac)` which corresponds to `ac`.
+
+# Examples
+```jldoctest
+julia> dP2 = del_pezzo(2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> d = ToricDivisor(dP2, [1,2,3,4,5])
+A torus-invariant, non-prime divisor on a normal toric variety
+
+julia> ac = AlgebraicCycle(d)
+An algebraic cycle on a normal toric variety given by 6V(x3)+V(e1)+7V(e2)
+
+julia> polynomial(ac)
+6*x3 + e1 + 7*e2
+```
+"""
+polynomial(ac::AlgebraicCycle) = ac.p
+export polynomial
+
+
+@doc Markdown.doc"""
+    polynomial(ring::MPolyQuo, ac::AlgebraicCycle)
+
+Returns the polynomial in `ring` corresponding
+to the algebraic cycle `ac`.
+
+# Examples
+```jldoctest
+julia> dP2 = del_pezzo(2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> d = ToricDivisor(dP2, [1,2,3,4,5])
+A torus-invariant, non-prime divisor on a normal toric variety
+
+julia> ac = AlgebraicCycle(d)
+An algebraic cycle on a normal toric variety given by 6V(x3)+V(e1)+7V(e2)
+
+julia> R, _ = PolynomialRing(QQ, 5)
+(Multivariate Polynomial Ring in x1, x2, x3, x4, x5 over Rational Field, fmpq_mpoly[x1, x2, x3, x4, x5])
+
+julia> (x1,x2,x3,x4,x5) = gens(R)
+5-element Vector{fmpq_mpoly}:
+ x1
+ x2
+ x3
+ x4
+ x5
+
+julia> sr_and_linear_relation_ideal = ideal([x1*x3, x1*x5, x2*x4, x2*x5, x3*x4, x1 + x2 - x5, x2 + x3 - x4 - x5])
+ideal(x1*x3, x1*x5, x2*x4, x2*x5, x3*x4, x1 + x2 - x5, x2 + x3 - x4 - x5)
+
+julia> R_quo = quo(R, sr_and_linear_relation_ideal)[1]
+Quotient of Multivariate Polynomial Ring in x1, x2, x3, x4, x5 over Rational Field by ideal(x1*x3, x1*x5, x2*x4, x2*x5, x3*x4, x1 + x2 - x5, x2 + x3 - x4 - x5)
+
+julia> polynomial(R_quo, ac)
+6*x3 + x4 + 7*x5
+```
+"""
+function polynomial(ring::MPolyQuo, ac::AlgebraicCycle)
+    p = polynomial(ac)
+    if iszero(p)
+        return zero(ring)
+    end
+    coeffs = [k for k in coefficients(p.f)]
+    expos = matrix(ZZ,[k for k in exponent_vectors(p.f)])
+    indets = gens(ring)
+    monoms = [prod(indets[j]^expos[k,j] for j in 1:ncols(expos)) for k in 1:nrows(expos)]
+    return sum(coeffs[k]*monoms[k] for k in 1:length(monoms))
```

Comment:
I think this code is essentially mapping the polynomial `p.f` into the ring `ring`, right? I think we have shorter & more efficient ways for doing this (and if not, we should). I am not at my computer right now and don't want to say something wrong right now (and I am not completely sure what type `p` resp. `p.f` have), but that target ring is of type `MPolyQuo`. Perhaps @fieker can quickly say how one is supposed to convert/map these MPolys

Reply:
Ah, I think `p` is also an `MPolyQuoElem`... isthe underlying polynomial ring supposed to have the "same" (identical? equivalent? isomorphic?) coefficient ring? Is it a good idea to blindly map variables by their index? I think the documentation for this function should perhaps explain what the word "corresponding" means in the sentence
> Returns the polynomial in `ring` corresponding to the algebraic cycle `ac`.

Reply:
> I think this code is essentially mapping the polynomial `p.f` into the ring `ring`, right? I think we have shorter & more efficient ways for doing this (and if not, we should). I am not at my computer right now and don't want to say something wrong right now (and I am not completely sure what type `p` resp. `p.f` have), but that target ring is of type `MPolyQuo`. Perhaps @fieker can quickly say how one is supposed to convert/map these MPolys

Yes that is the purpose of these lines. I am happy to hear how this can be done better/more efficiently.

Reply:
> Ah, I think `p` is also an `MPolyQuoElem`... isthe underlying polynomial ring supposed to have the "same" (identical? equivalent? isomorphic?) coefficient ring? Is it a good idea to blindly map variables by their index? I think the documentation for this function should perhaps explain what the word "corresponding" means in the sentence
> 
> > Returns the polynomial in `ring` corresponding to the algebraic cycle `ac`.

Ok. I see how this can be confusing. I have added more details here: https://github.com/oscar-system/Oscar.jl/pull/1449/commits/7be42e45269cd42bde286d9d5d7c6fbd41691cab

General philosophy:
- Algebraic cycle: Formal linear combination (over the integers) of irreducible subvarieties. (It makes sense to consider those with rational coefficients too. This is why scalar multiplication with `fmpq` is allowed. But one thing at a time.)
- For a well-defined intersection theory, need to consider them up to a certain equivalence relation. This is the so-called rational equivalence. (Or if you prefer: The ability, to move the cycles in "general position").

For simplicial, complete toric varieties, the algebraic counterpart looks as follows:
- Algebraic cycles are (isomorphic to) polynomials in the Cox ring S (Z^n graded polynomial ring over the complex numbers).
- Rational equivalence means to take a quotient of the Cox ring by a certain ideal I (the sum of the Stanley-Reisner ideal and the ideal of linear relations).
- In an abuse of terminology, I use "algebraic cycle" to refer to an equivalence class modulo rational equivalence.
- To get a "feeling" for such an equivalence class, I represent it by an algebraic cycle. In order words, to look at an equivalence class x in S/I I pick y in S such that the projection S -> S/I maps y to x.
- Consequently (as you already pointed out above - thank you for looking it at it carefully!), various functions depend on this choice of y. This includes the functions `coefficients`, `exponents`, `polynomial` as well as the display method.

Is it clear now? If not, I will go over the documentation again, add more details and try to make it better (suggestion welcome).